wdl/src/main/scala/wdl/WdlGraphNode.scala

```diff
@@ -72,15 +71,16 @@ object WdlGraphNode {
       }.toMap
     )
 
-    def foldFunction(acc: ErrorOr[FoldState], node: WdlGraphNode): ErrorOr[FoldState] = acc flatMap { goodAcc =>  buildNode(goodAcc, node) }
+    def foldFunction(acc: ErrorOr[FoldState], node: WdlGraphNode): ErrorOr[FoldState] = acc flatMap { goodAcc => buildNode(goodAcc, node) }
 
     def foldInGeneratedNodeAndNewInputs(acc: FoldState, outputPortPrefix: String)(gnani: GeneratedNodeAndNewNodes): FoldState = {
       // The output ports from the new node
       val newCallOutputPorts = (gnani.node.outputPorts map { p => s"$outputPortPrefix${p.name}" -> p }).toMap
       // The output ports from newly created GraphInputNodes:
       val newInputOutputPorts = (gnani.newInputs map { i => i.localName -> i.singleOutputPort }).toMap
+      val usedOuterGraphInputNodes = gnani.nestedOuterGraphInputNodes.map(_.singleUpstreamNode)
```

Comment:
I feel like I have no idea what I'm talking about but trying to understand this I would have assumed those would be added when the outer graph is constructed ?